jupyterhub/auth.py

```diff
@@ -37,6 +37,25 @@ class Authenticator(LoggingConfigurable):
     """Base class for implementing an authentication provider for JupyterHub"""
 
     db = Any()
+    
+    enable_auth_state = Bool(False, config=True,
+        help="""Enable persisting auth_state (if available).
+
+        auth_state will be encrypted and stored in the Hub's database.
+        This can include things like authentication tokens, etc.
+        to be passed to Spawners as environment variables.
+
+        Encrypting auth_state requires the cryptography package.
+
+        Additionally, the JUPYTERHUB_CRYPTO_KEY envirionment variable must
```

Comment:
True - we can revisit later if we introduce more of these!
 
re: other keys, yes, depending on the sensitivity of stuff we wanna encrypt, we *should* allow users to use different keys that might be rotated at different intervals, for example. However, I can't think of anything else we might wanna encrypt, so this is good!